src/db/migrations/1762074152438-AddPartsValidations.ts

```diff
@@ -0,0 +1,448 @@
+              CHECK ("product_id" ~ '^[A-Za-z]{1}[A-Za-z0-9_]{0,37}$'),
+          CONSTRAINT "base_product version"
+              CHECK ("product_version" ~ '^[1-9]\\d*(\\.(0|[1-9]\\d?))?$'),
+          CONSTRAINT "base_imaging time begin utc"
+              CHECK ("imaging_time_begin_utc" < now()),
+          CONSTRAINT "base_imaging time end utc"
+              CHECK ("imaging_time_end_utc" < now()),
+          CONSTRAINT "imaging times" CHECK (
+              "imaging_time_begin_utc" <= "imaging_time_end_utc"),
+          CONSTRAINT "base_resolution degree"
+              CHECK ("resolution_degree" BETWEEN 0.000000167638063430786 AND 0.703125),
+          CONSTRAINT "base_resolution meter"
+              CHECK ("resolution_meter" BETWEEN 0.0185 AND 78271.52),
+          CONSTRAINT "base_source resolution meter"
+              CHECK ("source_resolution_meter" BETWEEN 0.0185 AND 78271.52),
+          CONSTRAINT "base_horizontal accuracy ce90"
+              CHECK ("horizontal_accuracy_ce90" BETWEEN 0.01 AND 4000),
+          CONSTRAINT "base_parts_pkey" PRIMARY KEY ("id")
+      );
+    `);
+
+    // --- Validation table ---
+    await queryRunner.query(`
+      CREATE TABLE IF NOT EXISTS "polygon_parts"."validation_parts" (
+          "validated" boolean NOT NULL DEFAULT false,
+          "footprint" geometry(Geometry, 4326) NOT NULL,
+          CONSTRAINT "validation_footprint_type_chk"
+              CHECK (GeometryType("footprint") IN ('POLYGON','MULTIPOLYGON')),
+          CONSTRAINT "validation_footprint_srid_chk"
+              CHECK (ST_SRID("footprint") = 4326),
+          CONSTRAINT "validation_geometry_extent"
+              CHECK (Box2D("footprint") @ Box2D(ST_GeomFromText('LINESTRING(-180 -90, 180 90)'))),
+          CONSTRAINT "validation_parts_pkey" PRIMARY KEY ("id")
+      )
+      INHERITS ("polygon_parts"."base_parts");
+    `);
+
+    // --- Validation indexes ---
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_footprint_idx"
+      ON "polygon_parts"."validation_parts" USING GIST ("footprint");
+    `);
+
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_validated_idx"
+      ON "polygon_parts"."validation_parts" ("validated");
+    `);
+
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_id_idx"
+      ON "polygon_parts"."validation_parts" ("id");
+    `);
+
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_ingestion_date_utc_idx"
+      ON "polygon_parts"."validation_parts" ("ingestion_date_utc");
+    `);
+
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_imaging_time_begin_utc_idx"
+      ON "polygon_parts"."validation_parts" ("imaging_time_begin_utc");
+    `);
+
+    await queryRunner.query(`
+      CREATE INDEX IF NOT EXISTS "validation_parts_imaging_time_end_utc_idx"
+      ON "polygon_parts"."validation_parts" ("imaging_time_end_utc");
```

Comment:
i think this constraint could be removed since "footprint" is defined with srid of 4326. postgis would prevent inserting/updating with other SRID